src/path.cr

```diff
@@ -648,6 +648,9 @@ struct Path
   # ```
   def join(parts : Enumerable) : Path
     if parts.is_a?(Indexable)
+      # If it's just a single part we can avoid one allocation of String.build
+      return join_one(parts.first) if parts.size == 1
```

Comment:
Would it be cheaper on the optimizer to add a `def join(part : String | Path)` and delegate immediately to `join_one`?